wiki/Beatmapping/Mapping_techniques/Time-distance_equality/en.md

```diff
@@ -1,12 +1,31 @@
 ---
-stub: true
 tags:
   - distance snap
   - DS
 ---
 
 # Time-distance equality
 
-**Time-distance equality** is an [osu!](/wiki/Game_mode/osu!) [mapping technique](/wiki/Beatmapping/Mapping_techniques) where spacing between [hit objects](/wiki/Gameplay/Hit_object) on the playfield is proportional to the time between them. This is automatically calculated by the [distance snap](/wiki/Client/Beatmap_editor/Distance_snap) feature of the [beatmap editor](/wiki/Client/Beatmap_editor).
+::: Infobox
+![](/wiki/shared/editor/editor-draw-distsnap.png "The distance snap tool icon")
+:::
 
-In lower [difficulties](/wiki/Beatmap/Difficulty), time-distance equality is used to make gameplay more intuitive for new players.
+**Time-distance equality** is an [osu!](/wiki/Game_mode/osu!) [mapping technique](/wiki/Beatmapping/Mapping_techniques) where spacing between [hit objects](/wiki/Gameplay/Hit_object) on the playfield is proportional to the time between them. In the beatmap editor, it may be enforced by using the [distance snap](/wiki/Client/Beatmap_editor/Distance_snap) tool, which maintains the proportion according to the distance spacing multiplier.
+
+## Usage
+
+The equality principle is built around the idea that visual density of hit objects should correspond to the song's structure: intensive rhythms mean that objects should be located near each other, and vice versa. Applying time-distance equality makes the beatmap more readable and often intuitive, since it provides consistent visual cues as to when hit objects should be clicked.
```

Comment:
```suggestion
The equality principle is built around the idea that visual density of hit objects should correspond to the song's structure: faster rhythm means that objects should be located near each other, and vice versa. Applying time-distance equality makes the beatmap more readable and often intuitive, since it provides consistent visual cues as to when hit objects should be clicked.
```

Reply:
shouldn't this have been "a faster rhythm" or "faster rhythms"?